runtime/common/src/xcm_impl.rs

```diff
@@ -35,12 +37,19 @@ use xcm_executor::{
 	Assets,
 };
 
-pub fn native_currency_location(para_id: u32, id: CurrencyId) -> MultiLocation {
+pub fn local_currency_location(key: Vec<u8>) -> MultiLocation {
+	MultiLocation::new(
+		0,
+		X1(GeneralKey(WeakBoundedVec::<u8, ConstU32<32>>::force_from(key, None))),
+	)
+}
+
+pub fn native_currency_location(para_id: u32, key: Vec<u8>) -> MultiLocation {
```

Comment:
Vec<u8> as parameter can also deal with sibling currency, such as https://github.com/AcalaNetwork/Acala/blob/fd5d072a4670b02bf262c0468cb7d11e137ec71d/runtime/karura/src/xcm_config.rs#L355-L361. otherwise we need another method to simplify it.